File: Algorithms/result_manager_col.py
# Author: Mathieu Pont

from config import result_path, dataset_path
import pandas as pd
import sys
from statistics import mean, stdev

datasets = ["classic3", "classic4", "ng5", "ng20", "r8", "r40", "r52", "webkb"]
algos = ["CoclustInfo", "CoclustMod", "CoclustSpecMod"]

alphas = [0.9, 0.75, 0.5]

for dataset in datasets:
    print(dataset)
    base_file = result_path+"/best/"+dataset
    res = {i: [] for i in range(3)}
    for algo in algos:
        res_file = base_file+"_"+algo+"_col.txt"
        df = pd.read_csv(res_file, header=None)
        for i in range(3):
            res_i = str(round(mean(df.iloc[:,i]), 2))
            std = round(stdev(df.iloc[:,i]), 2)
            if std != 0.:
                res_i += " $\pm$ " + str(std)
            res[i].append(res_i)
    for name in res:
        post = "& $\\alpha=" + str(alphas[name]) + "$" #& $\alpha=0.75$
        print(post + " & " + ' & '.join(res[name]) + " \\\\")

File: Algorithms/result_manager_col.R
setwd(dirname(rstudioapi::getSourceEditorContext()$path))

library(ggplot2)
library(svglite)

best_result_path <- "../Results/best/"

datasets <- c("classic3", "classic4", "ng5", "ng20", "r8", "r40", "r52", "webkb")
algos <- c("CoclustInfo", "CoclustMod", "CoclustSpecMod")

plot_grouped_barplot <- function(data){
  x_names <- data[,1]
  Score <- data[,2]
  value <- as.double(data[,3])
  data <- data.frame(data)  
  my_barplot <- ggplot(data, aes(fill=Score, y=value, x=x_names)) + xlab("Algorithm") + ylab("Score") +
    geom_bar(position="dodge", stat="identity") + ylim(0, 1.0) + theme_grey(base_size = 15)
  print(my_barplot)
  return(my_barplot)
}

for(dataset in datasets){
  cat("##############\n#", dataset, "\n##############\n")
  words_path <- paste(best_result_path, dataset, "_words", sep = "")
  res_algos <- c()
  for(algo in algos){
    algo_res_col_path <- paste(best_result_path, dataset, "_", algo, "_col.txt", sep = "")
    algo_res_col <- mean(as.matrix(read.table(algo_res_col_path, sep=","))[,1])
    
    algo_res_row_path <- paste(best_result_path, "row_results/", dataset, "_", algo, ".txt", sep = "")
    algo_res_row <- mean(as.matrix(read.table(algo_res_row_path, sep=",")))
    
    res_algos <- rbind(res_algos, c(algo, "Row Score", algo_res_row))
    res_algos <- rbind(res_algos, c(algo, "Column Score", algo_res_col))
  }
  print(res_algos)
  svg_file_name <- paste(best_result_path, dataset, ".svg", sep="")
  # svg(filename = svg_file_name)
  # plot_grouped_barplot(res_algos)
  # dev.off()
  my_barplot <- plot_grouped_barplot(res_algos)
  print(my_barplot)
  break
  #ggsave(svg_file_name, device = "svg")
}

# create a dataset
specie <- c(rep("sorgho" , 3) , rep("poacee" , 3) , rep("banana" , 3) , rep("triticum" , 3) )
condition <- rep(c("normal" , "stress" , "Nitrogen") , 4)
value <- abs(rnorm(12 , 0 , 15))
data <- data.frame(specie,condition,value)

# Grouped
ggplot(data, aes(fill=condition, y=value, x=specie)) + 
  geom_bar(position="dodge", stat="identity")
